templates/glossary.md

```diff
@@ -0,0 +1,331 @@
+
+### binder
+
+Expressions such as `(a : α)`, `[a : α]` or `{a : α}` for any identifier(s) `a` and type `α` which, as part of various Lean syntactical elements -- [declarations](#declaration), `fun`, quantifiers and others -- represent identifiers which will be bound within the body of the syntactic element or declaration.
+
+Each type of binder has different implications for whether it will be bound implicitly (without being passed by a caller), explicitly, or via [typeclass inference](#typeclass-inference).
+
+In some places, notably within a `def`, defining "simple" binders without surrounding brackets are allowed, such as a binder `a` (with no explicit type) for some identifier `a`.
+
+### bundled vs unbundled
+
+Given a mathematical object `O` with property `P`, bundling `P` refers to the creation of a Lean [structure](#structure) containing a proof of `P` as one of its fields in addition to those needed to structurally define `O`.
+
+In contrast, an unbundled structure instead contains only the definition of `O`, and separately the creation of an `is_P` proposition which can be applied to terms of `O`.
+
+As a concrete example, a [group homomorphism](https://en.wikipedia.org/wiki/Group_homomorphism) can be viewed as a map `φ: G → H` between groups, along with a proof `h : φ(a * b) = φ(a) * φ(b)`. A bundled group homomorphism would contain both `φ` and `h` as fields, whereas an unbundled one would instead consist only of `φ`, with a separate `is_group_homomorphism` [declaration](#declaration) for the proof of `h`.
+
+There are performance, stylistic or implementation-related reasons to prefer bundling or unbundling.
+Generally within [mathlib](#mathlib), bundled structures are preferred, but unbundled versions are often also present.
+
+### cache
+
+Normally, a reference to a shared pre-built set of [`olean` files](#olean-file) built by [mathlib](#mathlib)'s [continuous integration](#continuous-integration) each time a commit is pushed to its repository.
+
+Its purpose is to alleviate the need for each [mathlib](#mathlib) user to build (or rebuild) the same Lean files locally, when doing so can take a significant number of time (multiple hours on a modest computer).
+
+The cache is built within the aforementioned continuous integration, and normally users of mathlib retrieve its built files using [`leanproject`](#leanproject).
+
+### `calc` mode
+
+A [mode](#mode) which consists of sequences of successive transformations of expressions involving a transitive relation such as `=`, `<` or others tagged with `trans` [attributes](#attribute).
+It is entered via the `calc` keyword.
+
+#### See also
+
+[The `calc` mode community documentation](https://leanprover-community.github.io/extras/calc.html)
+
+### carrier
+
+For a [bundled](#bundled-vs-unbundled) mathematical subobject on a type `T`, it is the *set* of [terms of `T`](#term) considered within the subobject.
```

Comment:
I don't think "carrier" has any particular connection to subobjects, does it? If we used fully bundled groups, the type of elements would (could) be called `carrier`. It would make mathematical sense to call `T` in `[group T]` the carrier type. https://en.wikipedia.org/wiki/Algebraic_structure